### PR TITLE
pyxtensor-feedstock

### PR DESCRIPTION
**Pull request by tdegeus**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pyxtensor-feedstock/build.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake -G "NMake Makefiles" -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% %SRC_DIR%
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/build.bat]: Please use `Ninja` or `NMake Makefiles JOM` for parallelism.

File: recipes/pyxtensor-feedstock/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cmake -DCMAKE_INSTALL_PREFIX=$PREFIX $SRC_DIR -DCMAKE_INSTALL_LIBDIR=lib
+make install
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/build.sh]:
```suggestion
make -j${CPU_COUNT} install
```

File: recipes/pyxtensor-feedstock/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  sha256: 3b1c7f0f1d5013aff15f3a78361ca6d5a0f8f9ceacc258d703fa46fa801dcbdf
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - cmake
+    - setuptools
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/meta.yaml]: This should only go into `host`
```suggestion
```

File: recipes/pyxtensor-feedstock/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  host:
+    - python
+    - pip
+    - setuptools
+  run:
+    - python
+    - setuptools
+    - pkg-config
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/meta.yaml]: Are really both `pkg-config` spelling needed?

File: recipes/pyxtensor-feedstock/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+source:
+  url: https://github.com/tdegeus/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 3b1c7f0f1d5013aff15f3a78361ca6d5a0f8f9ceacc258d703fa46fa801dcbdf
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/meta.yaml]: Probably needs compilers
```suggestion
  build:
   - {{ compiler('c') }}
```

Comment (tdegeus) [reply recipes/pyxtensor-feedstock/meta.yaml]: I added a C++ compiler

File: recipes/pyxtensor-feedstock/build.bat (hunk 1 above)
Comment (tdegeus) [reply recipes/pyxtensor-feedstock/build.bat]: Do I have to specify the number of CPUs also with `nmake` (like with `make` for unix-based systems)?

**Comment by tdegeus** [comment]
@xhochy Thank you for your useful comments! 

Regarding:

> I'm wondering whether this can actually be `noarch: python` as it is a header-only package.

I'm not sure here, if I'd use `noarch: python` would all the cmake and pkg-config configuration files be still set correctly on unix/windows?

**Comment by tdegeus** [comment]
@xhochy Another thing is that the CI seems to struggle to download the code, while the URL is standard and seems to be correct (at least I can download from the URL).

**Comment by xhochy** [comment]
> @xhochy Thank you for your useful comments!
> 
> Regarding:
> 
> > I'm wondering whether this can actually be `noarch: python` as it is a header-only package.
> 
> I'm not sure here, if I'd use `noarch: python` would all the cmake and pkg-config configuration files be still set correctly on unix/windows?

No, it wouldn't. This is a reason to not make this noarch.

File: recipes/pyxtensor-feedstock/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+{% set name = "pyxtensor" %}
+{% set version = "0.0.6" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/tdegeus/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (xhochy) [inline recipes/pyxtensor-feedstock/meta.yaml]: Small typo in the URL
```suggestion
  url: https://github.com/tdegeus/{{ name }}/archive/v{{ version }}.tar.gz
```

**Comment by tdegeus** [comment]
@xhochy Thanks for your comment. For this package it is then imperative not to use noarch. Another thing is that the CI finds a different checksum than I find on my system.

On my system:
```bash
openssl sha256 pyxtensor-0.0.6.tar.gz
SHA256(pyxtensor-0.0.6.tar.gz)= 3b1c7f0f1d5013aff15f3a78361ca6d5a0f8f9ceacc258d703fa46fa801dcbdf
```
while the CI finds
```none
dec5169eab7323a0de5993d93657896e4e6c76e5163475e70fdfa4b6c7e3c3ca
```

Why does this happen?

**Comment by xhochy** [comment]
I also get locally `dec5169eab7323a0de5993d93657896e4e6c76e5163475e70fdfa4b6c7e3c3ca`

**Comment by tdegeus** [comment]
@xhochy I fixed the build failures, or at least... The tests fail on windows, and I really cannot figure out why. Thanks (once more) for your help!

**Pull request merged**